### PR TITLE
Dark theme

### PR DESCRIPTION
**Pull request by xypnox**
A working dark theme and switcher have been added, also, the search input has been created but currently does nothing.

Screenshot_2020-04-30 LTTKGP 🎶 🎶 🎶(3)
Screenshot_2020-04-30 LTTKGP 🎶 🎶 🎶(2)

### CODE REVIEW

**Comment by ghostwriternr** [comment]
I'll take a proper look a bit later in the night. Quick first note: Would it be better to keep the dark mode icon somewhere in the sidebar (bottom-left)? Right now, have to come to the home page to switch each time. Feels a bit awkward sitting next to the search input.

**Comment by xypnox** [comment]
@ghostwriternr I have moved the switch to the sidebar and tweaked some parameters to make it more consistent.

**Comment by ghostwriternr** [comment]
One 'hack' is to use [React Refs](https://reactjs.org/docs/refs-and-the-dom.html), but I'm not sure it's the correct solution either. As mentioned in the link from official docs, this would lock the state within the component, yet again, stopping us from making use of Redux.

File: src/components/sidebar.tsx [hunk 1]
```diff
@@ -3,21 +3,53 @@ import * as Icon from 'react-feather';
 
 import { NavLink } from 'react-router-dom';
 
-type SidebarProps = {
-  expand: Boolean;
-  changeExpand: Function;
+let changeTheme = () => {
```
[the remark this answers is not in the document]
Comment (ghostwriternr) [reply src/components/sidebar.tsx:6]: https://programmingwithmosh.com/react/multiple-css-classes-react/

**Comment by xypnox** [comment]
Hi, I have preferred to keep the changing of theme off of react for the same reason that we were encountering the reload of video with the sidebar. 

The problem with keeping the classes in the state of the Application is that when the state changes (the state for themes, sidebar's short state) are stored either in the App component or the redux store. Both of which when changed apply a rerender of the entire app (as the theme is set in the html element or the topmost div.

We could escape this by making video a pure component however, I do not feel that changing the class names without using react's virtual dom will cause us noticeable issues. For example, the changes for both the sidebar and the theme are cosmetic, and for now, I do not find any need to store it in the state.

Also, these changes would not be as critical performance wise because, almost all of the users would set the theme once or twice during the entire use of the application, and same goes for the state of the sidebar.

I understand that changing the DOM without interacting with the virtual DOM is bad, but in this specific scenario, I find it easier to break the rule for a simpler code structure.

**Comment by xypnox** [comment]
And the problem with react rerendering the DOM is that the video component reloads the videos from the start.

**Pull request merged**